### PR TITLE
Variant alignment omitted words migration

### PR DESCRIPTION
**Pull request by fsimonjetz**

### CODE REVIEW

File: ebl/corpus/application/lines_updater.py [hunk 1]
```diff
@@ -9,10 +9,10 @@
 
 
 class LinesUpdater(ChapterUpdater):
-    def __init__(self, lines: LinesUpdate, sing_repository: SignRepository):
+    def __init__(self, lines: LinesUpdate, sign_repository: SignRepository):
```
Comment (fsimonjetz) [inline ebl/corpus/application/lines_updater.py:12]: spelling errors

File: ebl/corpus/application/schemas.py [hunk 2]
```diff
@@ -14,7 +14,9 @@
     Chapter,
     Classification,
 )
-from ebl.corpus.domain.line import Line, LineVariant, ManuscriptLine
+from ebl.corpus.domain.line import Line
+from ebl.corpus.domain.manuscript_line import ManuscriptLine
```
Comment (fsimonjetz) [inline ebl/corpus/application/schemas.py:18]: Extracted `ManuscriptLine` and `LineVariant` into individual files

File: ebl/corpus/application/signs_updater.py [hunk 3]
```diff
@@ -11,8 +11,8 @@
 
 
 class SignsUpdater:
-    def __init__(self, sing_repository: SignRepository):
-        self._sing_repository = MemoizingSignRepository(sing_repository)
+    def __init__(self, sign_repository: SignRepository):
```
Comment (fsimonjetz) [inline ebl/corpus/application/signs_updater.py:14]: Spelling errors

File: ebl/corpus/domain/manuscript_line.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
+from typing import Optional, Sequence, Tuple, Union, cast, Callable
```
Comment (fsimonjetz) [inline ebl/corpus/domain/manuscript_line.py:1]: Extracted from ebl/corpus/domain/line.py

File: ebl/corpus/web/chapter_schemas.py [hunk 5]
```diff
@@ -12,7 +12,9 @@
     labels,
     manuscript_id,
 )
-from ebl.corpus.domain.line import Line, LineVariant, ManuscriptLine
+from ebl.corpus.domain.line import Line
```
Comment (fsimonjetz) [inline ebl/corpus/web/chapter_schemas.py:15]: Refactoring

File: ebl/corpus/web/display_schemas.py [hunk 6]
```diff
@@ -2,7 +2,9 @@
 import attr
 from marshmallow import Schema, fields
 from ebl.bibliography.domain.reference import Reference
-from ebl.corpus.domain.line import Line, ManuscriptLine, LineVariant
+from ebl.corpus.domain.line import Line
```
Comment (fsimonjetz) [inline ebl/corpus/web/display_schemas.py:5]: Refactoring

File: ebl/tests/corpus/test_chapter_merge.py [hunk 7]
```diff
@@ -83,7 +85,7 @@ def test_merge_manuscript_line(old, new, expected):
             LineVariant(
                 (
                     AkkadianWord.of(
-                        (ValueToken.of("buāru"),), has_variant_alignmnet=True
+                        (ValueToken.of("buāru"),), has_variant_alignment=True
```
Comment (fsimonjetz) [inline ebl/tests/corpus/test_chapter_merge.py:88]: spelling errors

File: ebl/tests/corpus/test_corpus.py [hunk 8]
```diff
@@ -278,9 +280,20 @@ def test_updating_alignment(
     user,
     when,
 ) -> None:
-    aligmnet = 1
+    alignment = 1
```
Comment (fsimonjetz) [inline ebl/tests/corpus/test_corpus.py:283]: spelling error

File: ebl/tests/corpus/test_extant_lines_schema.py [hunk 9]
```diff
@@ -1,5 +1,7 @@
 from ebl.corpus.domain.chapter import Chapter
-from ebl.corpus.domain.line import Line, LineVariant, ManuscriptLine
+from ebl.corpus.domain.line import Line
```
Comment (fsimonjetz) [inline ebl/tests/corpus/test_extant_lines_schema.py:2]: refactoring

File: ebl/tests/corpus/test_line_variant.py [hunk 10]
```diff
@@ -1,8 +1,13 @@
+from typing import Sequence
 import attr
 import pytest
 
-from ebl.corpus.domain.line import LineVariant, ManuscriptLine
-from ebl.tests.factories.corpus import ManuscriptLineFactory
+from ebl.corpus.domain.line import ManuscriptLine
```
Comment (fsimonjetz) [inline ebl/tests/corpus/test_line_variant.py:5]: refactoring

File: ebl/tests/corpus/test_line_variant.py [hunk 11]
```diff
@@ -95,7 +100,32 @@ def test_set_has_variant_aligment(word: AbstractWord, expected: bool) -> None:
         INTERTEXT,
     )
     expected_reconstruction = (
-        AkkadianWord.of((ValueToken.of("buāru"),), has_variant_alignmnet=expected),
+        AkkadianWord.of((ValueToken.of("buāru"),), has_variant_alignment=expected),
     )
     expected_variant = attr.evolve(line_variant, reconstruction=expected_reconstruction)
     assert line_variant.set_has_variant_aligment() == expected_variant
+
+
```
Comment (fsimonjetz) [inline ebl/tests/corpus/test_line_variant.py:108]: new test case with/without omitted words

File: ebl/tests/corpus/test_parse_chapter.py [hunk 12]
```diff
@@ -2,7 +2,9 @@
 
 import pytest
 
-from ebl.corpus.domain.line import Line, LineVariant, ManuscriptLine
+from ebl.corpus.domain.line import Line
```
Comment (fsimonjetz) [inline ebl/tests/corpus/test_parse_chapter.py:5]: refactorings

File: ebl/tests/factories/corpus.py [hunk 13]
```diff
@@ -2,7 +2,9 @@
 import pydash
 
 from ebl.corpus.domain.chapter import Chapter, Classification
-from ebl.corpus.domain.line import Line, LineVariant, ManuscriptLine
+from ebl.corpus.domain.line import Line
```
Comment (fsimonjetz) [inline ebl/tests/factories/corpus.py:5]: refactorings

File: ebl/transliteration/domain/word_tokens.py [hunk 14]
```diff
@@ -94,7 +98,7 @@ def update_alignment(self: A, alignment_map) -> A:
             variant=None if new_alignment is None else self.variant,
         )
 
-    def merge(self, token: T) -> T:
+    def merge(self, token: T) -> T:  # sourcery skip
```
Comment (fsimonjetz) [inline ebl/transliteration/domain/word_tokens.py:101]: cf. https://docs.sourcery.ai/Configuration/Customizing-Refactorings/

File: ebl/corpus/application/alignment_updater.py [hunk 15]
```diff
@@ -6,7 +6,9 @@
 from ebl.corpus.application.chapter_updater import ChapterUpdater
 from ebl.corpus.domain.alignment import Alignment, ManuscriptLineAlignment
 from ebl.corpus.domain.chapter import Chapter, ChapterItem
-from ebl.corpus.domain.line import Line, LineVariant, ManuscriptLine
+from ebl.corpus.domain.line import Line
```
Comment (fsimonjetz) [inline ebl/corpus/application/alignment_updater.py:9]: refactoring

File: ebl/corpus/application/lemmatization_updater.py [hunk 16]
```diff
@@ -6,7 +6,9 @@
 from ebl.corpus.application.chapter_updater import ChapterUpdater
 from ebl.corpus.application.lemmatization import ChapterLemmatization
 from ebl.corpus.domain.chapter import Chapter, ChapterItem
-from ebl.corpus.domain.line import Line, LineVariant, ManuscriptLine
+from ebl.corpus.domain.line import Line
```
Comment (fsimonjetz) [inline ebl/corpus/application/lemmatization_updater.py:9]: refactoring

File: ebl/corpus/application/lines_updater.py (hunk 1 above)
Comment (fsimonjetz) [inline ebl/corpus/application/lines_updater.py:12]: spelling errors

File: ebl/corpus/domain/chapter.py [hunk 17]
```diff
@@ -6,7 +6,8 @@
 
 import ebl.corpus.domain.chapter_validators as validators
 from ebl.corpus.domain.extant_line import ExtantLine
-from ebl.corpus.domain.line import Line, ManuscriptLine, ManuscriptLineLabel
+from ebl.corpus.domain.line import Line
```
Comment (fsimonjetz) [inline ebl/corpus/domain/chapter.py:9]: refactoring

File: ebl/corpus/domain/chapter_display.py [hunk 18]
```diff
@@ -3,7 +3,8 @@
 import attr
 
 from ebl.corpus.domain.chapter import ChapterId, Chapter
-from ebl.corpus.domain.line import Line, LineVariant
+from ebl.corpus.domain.line import Line
```
Comment (fsimonjetz) [inline ebl/corpus/domain/chapter_display.py:6]: refactoring

File: ebl/corpus/application/text_validator.py [hunk 19]
```diff
@@ -91,12 +93,10 @@ def _visit_chapter(self, chapter: Chapter) -> None:
         for line in chapter.lines:
             self.visit(line)
 
-        invalid_lines: Sequence[str] = [
+        if invalid_lines := [
```
Comment (fsimonjetz) [inline ebl/corpus/application/text_validator.py:96]: check where this comes from

File: ebl/tests/corpus/test_line_variant.py [hunk 20]
```diff
@@ -95,7 +100,32 @@ def test_set_has_variant_aligment(word: AbstractWord, expected: bool) -> None:
...
+@pytest.mark.parametrize(
+    "omitted_words,expected",
+    [
+        (tuple(), (False, False, False)),
+        ((1, 2), (False, True, True)),
+    ],
+)
+def test_set_has_omitted_aligment(
```
Comment (fsimonjetz) [inline ebl/tests/corpus/test_line_variant.py:116]: To do: fix spelling error

File: ebl/tests/corpus/test_line_variant.py [hunk 21]
```diff
@@ -95,7 +100,32 @@ def test_set_has_variant_aligment(word: AbstractWord, expected: bool) -> None:
...
+    manuscript_line = ManuscriptLineFactory.build(omitted_words=omitted_words)
+    line_variant = LineVariantFactory.build(
+        reconstruction=(WORD, WORD, WORD),
+        manuscripts=(manuscript_line,),
+    )
+
+    assert (
+        tuple(
+            token.has_omitted_alignment
+            for token in line_variant.set_has_omitted_aligment().reconstruction
```
Comment (fsimonjetz) [inline ebl/tests/corpus/test_line_variant.py:128]: to do: fix spelling

File: ebl/transliteration/domain/word_tokens.py [hunk 22]
```diff
@@ -81,6 +82,9 @@ def set_alignment(
     def set_has_variant_alignment(self: A, has_variant_alignment) -> A:
```
Comment (fsimonjetz) [inline ebl/transliteration/domain/word_tokens.py:82]: to do: add type

File: ebl/transliteration/domain/word_tokens.py [hunk 23]
```diff
@@ -81,6 +82,9 @@ def set_alignment(
...
         return attr.evolve(self, has_variant_alignment=has_variant_alignment)
 
+    def set_has_omitted_alignment(self: A, has_omitted_alignment) -> A:
```
Comment (fsimonjetz) [inline ebl/transliteration/domain/word_tokens.py:85]: to do: add type

**Review by khoidt — APPROVED** [review]

File: ebl/corpus/application/alignment_updater.py [hunk 24]
```diff
@@ -82,9 +82,9 @@ def _visit_line_variant(self, variant: LineVariant) -> None:
             self.visit(manuscript_line)
 
         self._variants.append(
-            attr.evolve(variant, manuscripts=tuple(self._manuscript_lines))
-            .set_has_variant_aligment()
-            .set_has_omitted_aligment()
+            attr.evolve(
+                variant, manuscripts=tuple(self._manuscript_lines)
+            ).set_alignment_flags()
```
Comment (fsimonjetz) [inline ebl/corpus/application/alignment_updater.py:87]: `.set_alignment_flags` is a shorthand wrapper for `.set_has_variant_alignment().set_has_omitted_alignment()` (which almost always get called together).

File: ebl/corpus/domain/manuscript_line.py [hunk 25]
```diff
@@ -78,14 +78,14 @@ def merge(self, other: "ManuscriptLine") -> "ManuscriptLine":
         merged_line = self.line.merge(other.line)
         return attr.evolve(other, line=merged_line)
 
-    def update_alignments(self, aligment_map: AlignmentMap) -> "ManuscriptLine":
+    def update_alignments(self, alignment_map: AlignmentMap) -> "ManuscriptLine":
```
Comment (fsimonjetz) [inline ebl/corpus/domain/manuscript_line.py:81]: spelling errors

**Pull request merged**